packages/react/src/components/ActionMenu/ActionMenu.test.tsx

```diff
@@ -0,0 +1,549 @@
+  expect(menu).not.toBeVisible();
+});
+
+test('should render menu when trigger is clicked', async () => {
+  const user = userEvent.setup();
+  render(<ActionMenu {...defaultProps} />);
+
+  const trigger = screen.getByRole('button', { name: 'Trigger' });
+  const menu = screen.queryByRole('menu', { hidden: true });
+
+  await user.click(trigger);
+  await waitFor(() => {
+    expect(screen.queryByRole('menu')).toBeVisible();
+    expect(trigger).toHaveAttribute('aria-expanded', 'true');
+    expect(menu).toHaveAccessibleName('Trigger');
+  });
+
+  const menuItems = screen.getAllByRole('menuitem');
+  expect(menuItems).toHaveLength(3);
+  expect(menuItems[0]).toHaveAccessibleName('Item 1');
+  expect(menuItems[1]).toHaveAccessibleName('Item 2');
+  expect(menuItems[2]).toHaveAccessibleName('Item 3');
+});
+
+test('should focus menu when opened', async () => {
+  const user = userEvent.setup();
+  render(<ActionMenu {...defaultProps} />);
+
+  await user.click(screen.getByRole('button', { name: 'Trigger' }));
+  await waitFor(() => {
+    expect(screen.getByRole('menu')).toHaveFocus();
+  });
+});
+
+test('should return focus to trigger when closed', async () => {
+  const user = userEvent.setup();
+  render(<ActionMenu {...defaultProps} />);
+
+  const button = screen.getByRole('button', { name: 'Trigger' });
+  await user.click(button);
+  await user.keyboard('{Escape}');
+
+  await waitFor(() => {
+    expect(button).toHaveFocus();
+  });
+});
+
+test('should close menu when an action list item is selected', async () => {
+  const user = userEvent.setup();
+  render(<ActionMenu {...defaultProps} />);
+
+  await user.click(screen.getByRole('button', { name: 'Trigger' }));
+  await user.click(screen.getByText('Item 2'));
+
+  await waitFor(() => {
+    expect(screen.queryByRole('menu', { hidden: true })).not.toBeVisible();
+  });
+});
+
+test('should close menu when clicked outside', async () => {
+  const user = userEvent.setup();
+  render(<ActionMenu {...defaultProps} />);
+
+  await user.click(screen.getByRole('button', { name: 'Trigger' }));
+  await user.click(document.body);
+
+  await waitFor(() => {
+    expect(screen.queryByRole('menu', { hidden: true })).not.toBeVisible();
+  });
+});
+
+test('should close menu on escape keypress', async () => {
+  const user = userEvent.setup();
+  render(<ActionMenu {...defaultProps} />);
+
+  await user.click(screen.getByRole('button', { name: 'Trigger' }));
+  await user.keyboard('{Escape}');
+
+  await waitFor(() => {
+    expect(screen.queryByRole('menu', { hidden: true })).not.toBeVisible();
+  });
+});
+
+test('should open menu on arrow down key press', async () => {
+  const user = userEvent.setup();
+  render(<ActionMenu {...defaultProps} />);
+
+  const triggerButton = screen.getByRole('button', { name: 'Trigger' });
+
+  triggerButton.focus();
+  await user.keyboard('{ArrowDown}');
+
+  await waitFor(() => {
+    expect(screen.queryByRole('menu')).toBeVisible();
+  });
```

Comment:
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-waitfor-to-wait-for-elements-that-can-be-queried-with-find

We can use `findByRole` here instead and Kent says the error message will be better.

```suggestion
expect(await screen.findByRole('menu')).toBeVisible();
```